### PR TITLE
Compute max call depth

### PR DESCRIPTION
**Pull request by eponier**
Proof of concept @bgregoir 
Should a function that calls no other function have 0 or 1 for "max call depth"?

### CODE REVIEW

**Comment by eponier** [comment]
Should be ready. It computes the max call depth at stack allocation time and puts this piece of information in stk_fun_extra. There is also an annotation if the user wants to specify an expected value.

**Review by vbgl — COMMENTED** [review]
Can you please add a test-case showing the annotation in use?

Also, this deserves an entry in the change log.

File: compiler/src/stackAlloc.ml [hunk 1]
```diff
@@ -213,15 +215,17 @@ let memory_analysis pp_err ~debug tbl up =
...
-          let fn_algin = sao.Stack_alloc.sao_align in
-          let align = if wsize_lt align fn_algin then fn_algin else align in
+          let fn_align = sao.Stack_alloc.sao_align in
+          let align = if wsize_lt align fn_align then fn_align else align in
           let fn_max = Conv.z_of_cz (sao.Stack_alloc.sao_max_size) in
           let max_stk = if Z.lt max_stk fn_max then fn_max else max_stk in
-          align, max_stk
-        ) sao.sao_calls (align, Z.zero) in
+          let fn_max_call_depth = Conv.z_of_cz (sao.Stack_alloc.sao_max_call_depth) in
+          let max_call_depth = if Z.lt max_call_depth fn_max_call_depth then fn_max_call_depth else max_call_depth in
```
[the remark this answers is not in the document]
Comment (eponier) [reply compiler/src/stackAlloc.ml]: done

**Comment by eponier** [comment]
I added a changelog entry, and two test cases: one succeeding, the other failing.

**Comment by vbgl** [comment]
Please add an export function in the successful example. (The compiler has unspecified behavior on programs without any export function.)

**Comment by eponier** [comment]
Done.

**Review by vbgl — APPROVED** [review]
Thanks. I’ll merge as soon as CI gets green.

**Pull request merged**